hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/replication/SimpleContainerDownloader.java

```diff
@@ -74,21 +75,13 @@ public SimpleContainerDownloader(ConfigurationSource conf,
     for (DatanodeDetails datanode : sourceDatanodes) {
       try {
         if (result == null) {
-          GrpcReplicationClient grpcReplicationClient =
-              new GrpcReplicationClient(datanode.getIpAddress(),
-                  datanode.getPort(Name.STANDALONE).getValue(),
-                  workingDirectory, securityConfig, caCert);
-          result = grpcReplicationClient.download(containerId);
+          result = downloadContainer(containerId, datanode);
         } else {
           result = result.thenApply(CompletableFuture::completedFuture)
               .exceptionally(t -> {
                 LOG.error("Error on replicating container: " + containerId, t);
                 try {
-                  GrpcReplicationClient grpcReplicationClient =
-                      new GrpcReplicationClient(datanode.getIpAddress(),
-                          datanode.getPort(Name.STANDALONE).getValue(),
-                          workingDirectory, securityConfig, caCert);
-                  return grpcReplicationClient.download(containerId);
+                  return downloadContainer(containerId, datanode);
                 } catch (IOException e) {
                   LOG.error("Error on replicating container: " + containerId,
                       t);
```

Comment:
Not introduced in this patch, but I think this should be:

```suggestion
                } catch (IOException e) {
                  LOG.error("Error on replicating container: " + containerId,
                      e);
```

Reply:
Nice catch, thx.